src/main/java/nl/tudelft/contextproject/Main.java

```diff
@@ -91,10 +121,13 @@ public static void setInstance(Main main) {
 	public boolean setController(Controller c) {
 		if (c != controller) {
 			if (controller != null) {
-				stateManager.detach(controller);
+				getStateManager().detach(controller);
 			}
 			controller = c;
-			stateManager.attach(controller);
+			
+			if (controller != null) {

```

Comment:
On line 123 the old controller is checked, on 128 the new controller (set on line 126)